### PR TITLE
Create codecov.yml

### PR DESCRIPTION
**Pull request by andrewkern**

### CODE REVIEW

**Comment by petrelharp** [comment]
What would this change - do you know? This is all a big black box to me.

**Comment by andrewkern** [comment]
so i think adding this yaml is automatically looked at by the codecov orb thingy. i'm not sure exactly where it should live though. it might go at the repo root directory?

**Comment by petrelharp** [comment]
Right - I'm wondering if you know what the effect on the behavior of codecov is going to be. How is it changing things?

**Comment by andrewkern** [comment]
oh i'm trying to move the target here to 95%. that's all.

File: .circleci/codecov.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+        enabled: yes           # must be yes|true to enable this status
+        target: 95             # specify the target coverage for each commit status
+                               #   option: "auto" (must increase from parent commit or pull request base)
+                               #   option: "X%" a static target percentage to hit
+        threshold: null        # allowed to drop X% and still result in a "success" commit status
+        if_not_found: success  # if parent is not found report status as success, error, or failure
+        if_ci_failed: error    # if ci fails report status as success, error, or failure
+    patch:
+      default:
+        enabled: no
```
Comment (petrelharp) [inline .circleci/codecov.yml:18]: any reason you're disabling `patch`?

**Comment by petrelharp** [comment]
Currently, what's the policy? Is it
```
                                #   option: "auto" (must increase from parent commit or pull request base)
```

If so, I guess I'd vote to actually keep it at `auto` but then we know that sometimes we'll need to merge PRs that don't "pass" codecov just because they've decreased the denominator (say).

In any case - right now it looks like we're at 99.52% covered, so 95% is a big drop?

File: .circleci/codecov.yml (hunk 1 above)
Comment (andrewkern) [reply .circleci/codecov.yml:18]: i think that's what it's supposed to be by default? no sure

**Comment by andrewkern** [comment]
don't know?

**Comment by andrewkern** [comment]
target: `auto` looks good actually: https://docs.codecov.com/docs/common-recipe-list

**Comment by petrelharp** [comment]
So - maybe we leave things the way they are and not merge this PR?

**Comment by jeromekelleher** [comment]
I had a look at the settings and it looks like we're on auto settings for codecov. The msprime config is [here](https://github.com/tskit-dev/msprime/blob/main/.codecov.yml) if you want a reference - I think the yaml file must be in the root directory, though.

To be honest, this is really quite fiddly and it would be simpler to bump the test coverage up than the change the settings.

**Comment by petrelharp** [comment]
> To be honest, this is really quite fiddly and it would be simpler to bump the test coverage up than the change the settings.

lol.

**Comment by andrewkern** [comment]
ugh okay! closing this PR

**Pull request closed without merging**